argus/src/main/java/com/moldedbits/argus/handler/BaseProviderHandler.java

```diff
@@ -0,0 +1,48 @@
+package com.moldedbits.argus.handler;
+
+import android.text.TextUtils;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.moldedbits.argus.ArgusTheme;
+import com.moldedbits.argus.R;
+import com.moldedbits.argus.provider.BaseProvider;
+import com.moldedbits.argus.utils.ThemeHelper;
+
+/*
+    Handler which will :-
+  * Apply Theme to SignIn and SignUp Providers
+  * Provides login functionality for specific end point
+ */
+public abstract class BaseProviderHandler extends BaseProvider implements ThemeHelper {
+
+    @Override
+    public void applyTheme(View view, ArgusTheme theme) {
+        TextView welcomeTv = (TextView) view.findViewById(R.id.tv_welcome_text);
+        if (welcomeTv != null && !TextUtils.isEmpty(theme.getWelcomeText())) {
+            welcomeTv.setText(theme.getWelcomeText());
+            welcomeTv.setTextSize(theme.getWelcomeTextSize());
```

Comment:
I haven't implemented the Size thing, it was from before, though I agree with your point